runtime-modules/bureaucracy/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+mod types;
+
+use sr_primitives::traits::One;
+use srml_support::traits::Currency;
+use srml_support::{decl_module, decl_storage, dispatch, ensure};
+use system::ensure_root;
+
+use membership::role_types::{ActorInRole, Role};
+use types::{Lead, LeadRoleState};
+
+//TODO: convert messages to the decl_error! entries
+pub static MSG_ORIGIN_IS_NOT_LEAD: &str = "Origin is not lead";
+pub static MSG_CURRENT_LEAD_NOT_SET: &str = "Current lead is not set";
+pub static MSG_CURRENT_LEAD_ALREADY_SET: &str = "Current lead is already set";
+pub static MSG_IS_NOT_LEAD_ACCOUNT: &str = "Not a lead account";
+
+/// Type identifier for lead role, which must be same as membership actor identifier
+pub type LeadId<T> = <T as membership::members::Trait>::ActorId;
+
+/// Type of minting reward relationship identifiers
+pub type RewardRelationshipId<T> = <T as recurringrewards::Trait>::RewardRelationshipId;
+
+/// Balance type of runtime
+pub type BalanceOf<T> =
+    <<T as stake::Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
+/// The bureaucracy main _Trait_
+pub trait Trait<I: Instance>:
+    system::Trait + recurringrewards::Trait + membership::members::Trait
+{
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait<I>, I: Instance> as Bureaucracy {
+        /// The current lead.
+        pub CurrentLeadId get(current_lead_id) : Option<LeadId<T>>;
+
+        /// Maps identifier to corresponding lead.
+        pub LeadById get(lead_by_id): linked_map LeadId<T> => Lead<T::AccountId, T::RewardRelationshipId, T::BlockNumber>;
+
+        /// Next identifier for new current lead.
+        pub NextLeadId get(next_lead_id): LeadId<T>;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait<I>, I: Instance> for enum Call where origin: T::Origin {
+    /// Introduce a lead when one is not currently set.
+    pub fn set_lead(origin, member: T::MemberId, role_account: T::AccountId) -> dispatch::Result {
+        ensure_root(origin)?;
+
+        // Ensure there is no current lead
+        ensure!(
+            <CurrentLeadId<T, I>>::get().is_none(),
+            MSG_CURRENT_LEAD_ALREADY_SET
+        );
+
+        let new_lead_id = <NextLeadId<T, I>>::get();
+
+        let new_lead_role = ActorInRole::new(Role::CuratorLead, new_lead_id);
+
+        //
+        // == MUTATION SAFE ==
+        //
+
+        // Register in role - will fail if member cannot become lead
+        membership::members::Module::<T>::register_role_on_member(member, &new_lead_role)?;
```

Comment:
All this registration stuff goes away, as membership module will not be relied on to hold member <-> role binding information.